File: src/main/java/me/windyteam/kura/module/modules/render/NoRender.java
package me.windyteam.kura.module.modules.render;

import me.windyteam.kura.event.events.client.PacketEvents;
import me.windyteam.kura.event.events.render.*;
import me.windyteam.kura.event.events.render.*;
import me.windyteam.kura.module.Category;
import me.windyteam.kura.module.Module;
import me.windyteam.kura.setting.BooleanSetting;
import me.windyteam.kura.setting.Setting;
import net.minecraft.block.BlockSnow;
import net.minecraft.block.state.IBlockState;
import net.minecraft.init.Blocks;
import net.minecraft.network.Packet;
import net.minecraft.network.play.server.*;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.tileentity.TileEntityEnchantmentTable;
import net.minecraft.tileentity.TileEntityEnderChest;
import net.minecraftforge.client.event.RenderBlockOverlayEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

/**
 * Created by 086 on 4/02/2018.
 * Updated by S-B99 on 14/12/19
 */
@Module.Info(name = "NoRender", category = Category.RENDER, description = "Ignore entity spawn packets")
public class NoRender extends Module {
    public static NoRender INSTANCE = new NoRender();
    public Setting<Skylight> skylight = msetting("SkyLight", Skylight.ALL);
    public final Setting<Boolean> BlockLayer = bsetting("BlockLayer", true);
    public final Setting<Boolean> mob = bsetting("Mob", false);
    public final Setting<Boolean> sand = bsetting("Sand", false);
    public final Setting<Boolean> gentity = bsetting("GEntity", false);
    public final Setting<Boolean> object = bsetting("Object", false);
    public final Setting<Boolean> xp = bsetting("XP", false);
    public final Setting<Boolean> paint = bsetting("Paintings", false);
    public final Setting<Boolean> fire = bsetting("Fire", true);
    public final Setting<Boolean> explosion = bsetting("Explosions", true);
    public BooleanSetting skylightupdate = bsetting("SkylightUpdate", true);
    public Setting<Boolean> totemPops = bsetting("Totem", false);
    public BooleanSetting table = bsetting("EnchantmentTable", false);
    public BooleanSetting enderChest = bsetting("EnderChest", false);
    public BooleanSetting banner = bsetting("Banner", false);

    public static NoRender getInstance() {
        if (INSTANCE == null)
            INSTANCE = new NoRender();
        return INSTANCE;
    }

    @SubscribeEvent
    public void oao(RenderBlockOverlayEvent event) {
        if (fire.getValue() && event.getOverlayType() == RenderBlockOverlayEvent.OverlayType.FIRE)
            event.setCanceled(true);
    }

    @SubscribeEvent
    public void onRenderOverlay(RenderOverlayEvent event) {
        event.setCanceled(true);
    }

    @SubscribeEvent
    public void RenderLight(RenderLightEvent event) {
        if (skylightupdate.getValue()) {
            event.setCanceled(true);
        }
    }

    @SubscribeEvent
    public void totemPop(RenderTotemPopEvent event) {
        if (totemPops.getValue()) {
            event.setCanceled(true);
        }
    }


    @SubscribeEvent
    public void banner(RenderBannerEvent event) {
        if (banner.getValue()) {
            event.setCanceled(true);
        }
    }

    @SubscribeEvent
    public void enderChest(RenderEnderChestEvent event) {
        if (enderChest.getValue()) {
            event.setCanceled(true);
        }
    }

    @SubscribeEvent
    public void enchantmentTable(RenderEnchantmentTableEvent event) {
        if (table.getValue()) {
            event.setCanceled(true);
        }
    }

    @SubscribeEvent
    public void BlockLayer(RenderLiquidVisionEvent event) {
        if (BlockLayer.getValue()) {
            event.setCanceled(true);
        }
    }

    @SubscribeEvent
    public void awa(PacketEvents.Receive event) {
        Packet<?> packet = event.getPacket();
        if ((packet instanceof SPacketSpawnMob && mob.getValue()) ||
                (packet instanceof SPacketSpawnGlobalEntity && gentity.getValue()) ||
                (packet instanceof SPacketSpawnObject && object.getValue()) ||
                (packet instanceof SPacketSpawnExperienceOrb && xp.getValue()) ||
                (packet instanceof SPacketSpawnObject && sand.getValue()) ||
                (packet instanceof SPacketExplosion && explosion.getValue()) ||
                (packet instanceof SPacketSpawnPainting && paint.getValue()))
            event.setCanceled(true);
    }

    public boolean tryReplaceEnchantingTable(TileEntity tileEntity) {
        if (table.getValue() && tileEntity instanceof TileEntityEnchantmentTable) {
            IBlockState blockState = Blocks.SNOW_LAYER.defaultBlockState.withProperty(BlockSnow.LAYERS, 7);
            mc.world.setBlockState(tileEntity.getPos(), blockState);
            mc.world.markTileEntityForRemoval(tileEntity);
            return true;
        }
        return false;
    }

    public boolean tryReplaceEnderChest(TileEntity tileEntity) {
        if (enderChest.getValue() && tileEntity instanceof TileEntityEnderChest) {
            IBlockState blockState = Blocks.SNOW_LAYER.defaultBlockState.withProperty(BlockSnow.LAYERS, 7);
            mc.world.setBlockState(tileEntity.getPos(), blockState);
            mc.world.markTileEntityForRemoval(tileEntity);
            return true;
        }
        return false;
    }

    public void setInstance() {
        INSTANCE = this;
    }

    public enum Skylight {
        NONE, WORLD, ENTITY, ALL
    }

}

File: src/main/java/me/windyteam/kura/module/modules/render/NoRender.kt
package me.windyteam.kura.module.modules.render

import me.windyteam.kura.event.events.client.PacketEvents
import me.windyteam.kura.event.events.render.*
import me.windyteam.kura.module.Category
import me.windyteam.kura.module.Module
import me.windyteam.kura.setting.Setting
import net.minecraft.block.BlockSnow
import net.minecraft.init.Blocks
import net.minecraft.network.Packet
import net.minecraft.network.play.server.*
import net.minecraft.tileentity.TileEntity
import net.minecraft.tileentity.TileEntityEnchantmentTable
import net.minecraft.tileentity.TileEntityEnderChest
import net.minecraftforge.client.event.RenderBlockOverlayEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

@Module.Info(name = "NoRender", category = Category.RENDER, description = "Ignore entity spawn packets")
object NoRender : Module() {

    @JvmField
    var skylight = msetting("SkyLight", Skylight.NONE)
    val BlockLayer: Setting<Boolean> = bsetting("BlockLayer", true)
    val mob: Setting<Boolean> = bsetting("Mob", false)
    val sand: Setting<Boolean> = bsetting("Sand", false)
    val gentity: Setting<Boolean> = bsetting("GEntity", false)
    val `object`: Setting<Boolean> = bsetting("Object", false)
    val xp: Setting<Boolean> = bsetting("XP", false)
    val paint: Setting<Boolean> = bsetting("Paintings", false)
    val fire: Setting<Boolean> = bsetting("Fire", true)
    val explosion: Setting<Boolean> = bsetting("Explosions", true)
    var skylightupdate = bsetting("SkylightUpdate", true)
    var totemPops: Setting<Boolean> = bsetting("Totem", false)
    var table = bsetting("EnchantmentTable", false)
    var enderChest = bsetting("EnderChest", false)
    var banner = bsetting("Banner", false)
    val armor: Setting<Boolean> = bsetting("Armor",true)
    @SubscribeEvent
    fun oao(event: RenderBlockOverlayEvent) {
        if (fire.value && event.overlayType == RenderBlockOverlayEvent.OverlayType.FIRE) event.isCanceled = true
    }

    @SubscribeEvent
    fun onRenderOverlay(event: RenderOverlayEvent) {
        event.isCanceled = true
    }

    @SubscribeEvent
    fun RenderLight(event: RenderLightEvent) {
        if (skylightupdate.value) {
            event.isCanceled = true
        }
    }

    @SubscribeEvent
    fun totemPop(event: RenderTotemPopEvent) {
        if (totemPops.value) {
            event.isCanceled = true
        }
    }

    @SubscribeEvent
    fun banner(event: RenderBannerEvent) {
        if (banner.value) {
            event.isCanceled = true
        }
    }

    @SubscribeEvent
    fun enderChest(event: RenderEnderChestEvent) {
        if (enderChest.value) {
            event.isCanceled = true
        }
    }

    @SubscribeEvent
    fun enchantmentTable(event: RenderEnchantmentTableEvent) {
        if (table.value) {
            event.isCanceled = true
        }
    }

    @SubscribeEvent
    fun BlockLayer(event: RenderLiquidVisionEvent) {
        if (BlockLayer.value) {
            event.isCanceled = true
        }
    }

    @SubscribeEvent
    fun awa(event: PacketEvents.Receive) {
        val packet = event.getPacket<Packet<*>>()
        if (packet is SPacketSpawnMob && mob.value || packet is SPacketSpawnGlobalEntity && gentity.value || packet is SPacketSpawnObject && `object`.value || packet is SPacketSpawnExperienceOrb && xp.value || packet is SPacketSpawnObject && sand.value || packet is SPacketExplosion && explosion.value || packet is SPacketSpawnPainting && paint.value) event.isCanceled =
            true
    }

    fun tryReplaceEnchantingTable(tileEntity: TileEntity): Boolean {
        if (table.value && tileEntity is TileEntityEnchantmentTable) {
            val blockState = Blocks.SNOW_LAYER.defaultBlockState.withProperty(BlockSnow.LAYERS, 7)
            mc.world.setBlockState(tileEntity.getPos(), blockState)
            mc.world.markTileEntityForRemoval(tileEntity)
            return true
        }
        return false
    }

    fun tryReplaceEnderChest(tileEntity: TileEntity): Boolean {
        if (enderChest.value && tileEntity is TileEntityEnderChest) {
            val blockState = Blocks.SNOW_LAYER.defaultBlockState.withProperty(BlockSnow.LAYERS, 7)
            mc.world.setBlockState(tileEntity.getPos(), blockState)
            mc.world.markTileEntityForRemoval(tileEntity)
            return true
        }
        return false
    }

    enum class Skylight {
        NONE,
        WORLD,
        ENTITY,
        ALL
    }

}
